src/modals/Deploy.tsx

```diff
@@ -48,18 +46,24 @@ function Deploy() {
     const handleEstimateDeploySmartWalletButtonClick = async () => {
         setEstimateLoading(true);
         try {
-            const opts: RelayGasEstimationOptions = {
-                abiEncodedTx: '0x',
-                destinationContract: ZERO_ADDRESS,
-                smartWalletAddress: smartWallet!.address,
-                tokenFees: '1',
-                isSmartWalletDeploy: true,
-                index: smartWallet!.index.toString(),
-                tokenAddress: token!.instance.address
+            const relayTransactionOpts: UserDefinedEnvelopingRequest = {
+                request: {
+                    from: account,
+                    value: '0',
+                    data: '0x00',
+                    to: token!.instance.address,
+                    tokenAmount: '0',
+                    tokenContract: token!.instance.address,
+                    validUntilTime: 0,
+                    index: smartWallet!.index
+                },
+                relayData: {
+                    callForwarder: process.env['REACT_APP_CONTRACTS_SMART_WALLET_FACTORY']!
+                }
             };
 
-            const estimation: RelayEstimation =
-                await provider!.estimateMaxPossibleGas(opts);
+            const estimation: RelayEstimation = await relayClient!.estimateTransaction(relayTransactionOpts, {});
```

Comment:
The `RelayClient` does not use the `RequestConfig`, I think that we should remove it and use it in the `RelayProvider`